x-pack/plugins/session_view/server/routes/process_events_route.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { schema } from '@kbn/config-schema';
+import type { ElasticsearchClient, Logger } from 'kibana/server';
+import { IRouter } from '../../../../../src/core/server';
+import { PROCESS_EVENTS_ROUTE, PROCESS_EVENTS_PER_PAGE } from '../../common/constants';
+import { expandDottedObject } from '../../common/utils/expand_dotted_object';
+
+export const registerProcessEventsRoute = (router: IRouter, logger: Logger) => {
+  router.get(
+    {
+      path: PROCESS_EVENTS_ROUTE,
+      validate: {
+        query: schema.object({
+          sessionEntityId: schema.string(),
+          cursor: schema.maybe(schema.string()),
+          forward: schema.maybe(schema.boolean()),
+        }),
+      },
+    },
+    async (context, request, response) => {
+      const client = context.core.elasticsearch.client.asCurrentUser;
+      const { sessionEntityId, cursor, forward = true } = request.query;
+      const body = await doSearch(client, sessionEntityId, cursor, forward);
+
+      return response.ok({ body });
+    }
+  );
+};
+
+export const doSearch = async (
+  client: ElasticsearchClient,
+  sessionEntityId: string,
+  cursor: string | undefined,
+  forward = true
+) => {
+  // Temporary hack. Updates .siem-signals-default index to include a mapping for process.entry_leader.entity_id
+  // TODO: find out how to do proper index mapping migrations...
+  let siemSignalsExists = true;
+
+  try {
+    await client.indices.putMapping({
+      index: '.siem-signals-default',
+      body: {
+        properties: {
+          'process.entry_leader.entity_id': {
```

Comment:
Seems like overkill as well, but if possible try to put the `process.entry_leader.entity_id` behind a constant as well. We just went through changes that renamed fields across the entire application. Using constants for these properties will make any of these changes much easier in the future. The file that was created for this effort is here: https://github.com/elastic/kibana/blob/main/packages/kbn-rule-data-utils/src/technical_field_names.ts